authors/apps/authentication/tests/test_base.py

```diff
@@ -0,0 +1,29 @@
+from rest_framework.test import APITestCase, APIClient
+from django.urls import reverse
+from ..models import User
+
+class TestBase(APITestCase):
+    
+    def setUp(self):
+        self.client = APIClient()
+        self.user_data = {
+                'user': {
+                    'email': "test@testuser.com",
+                    'username': "testuser",
+                    'password': "password"
+                    }
+                }
+        self.test_user = {
+            "user": {
+                "username": "neelxie",
+                "email": "derek@derek.com",
+                "password": "realers"
+            }
+        }
+        self.new_test_user = User.objects.create_user(username="neelxie",
+                                             email="derek@derek.com",
+                                             password="realers")
+        self.url = reverse("authentication:login")
+        self.register_url = reverse("authentication:register") 
+        # use the APIClient to log in the user to be tested against
+        self.client.login(email="derek@derek.com", password="realers")
```

Comment:
@neelxie I suggest you assign this line to a variable so that it can be accessed by the different test cases.

Reply:
@alexxsanya let me work on that